src/f32-rsum/rvv.c.in

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 Imagination Technologies, Inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$assert LMUL in [1, 2, 4, 8]
+#include <assert.h>
+
+#include <xnnpack/common.h>
+#include <xnnpack/reduce.h>
+#include <riscv_vector.h>
+
+void xnn_f32_rsum_ukernel__rvv_u${LMUL}v(
+    size_t batch,
+    const float* input,
+    float* output,
+    const union xnn_f32_scale_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  batch >>= XNN_LOG2_SIZEOF_FLOAT;
+  int32_t n = __riscv_vsetvl_e32m${LMUL}(batch);
+  vfloat64m1_t acc_f64v = __riscv_vfmv_s_f_f64m1(0.f, n);
+  do {
+    n = __riscv_vsetvl_e32m${LMUL}(batch);
+
+    vfloat32m${LMUL}_t in_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+    acc_f64v = __riscv_vfwredosum_vs_f32m${LMUL}_f64m1(in_f32v, acc_f64v, n);
```

Comment:
Could we try to use `several vector-add + one vector reduced sum` to implement it?
Intuitively, `vadd` can have better performance comparing to `reduced sum`. If we want to have similar performance between `vadd` & `reduced sum`, HW complexity will increase.
Hence, `several vector-add + 1 vector reduced sum` might be more general for different CPU.

Reply:
I was also wanting to ask about this. Is the ordered sum required to keep the same ordering of operations as in scalar? It could also be useful to do some unrolling, depending on which LMUL is used. An optimised LMUL=1 implementation will look different from an optimised LMUL=8 implementation

Reply:
I am not sure about the precision constraint in XNNPACK. Maybe @fbarchard can confirm?
Do we need to have the exact computation order?

Reply:
I do not think order of summation will matter in this case. I will try with unrolling logic.

Reply:
Unrolled the loops for different LMUL cases.